python/dgl/optim/pytorch/sparse_optim.py

```diff
@@ -647,12 +707,25 @@ def update(self, idx, grad, emb):
 
             update_mem = beta1 * orig_mem + (1.-beta1) * grad_mem
             update_power = beta2 * orig_power + (1.-beta2) * grad_power
-            update_mem_dst = update_mem.to(state_dev, non_blocking=True)
-            update_power_dst = update_power.to(state_dev, non_blocking=True)
-            if state_block:
-                # use events to try and overlap CPU and GPU as much as possible
-                update_event = th.cuda.Event()
-                update_event.record()
+
+            if use_uva:
+                scatter_pinned_tensor_rows(state_mem, \
+                                           grad_indices, \
+                                           update_mem.to(dtype=self._dtype))
+                scatter_pinned_tensor_rows(state_power, \
+                                           grad_indices, \
+                                           update_power.to(dtype=self._dtype))
+            else:
+                update_mem_dst = update_mem.to(state_dev, \
+                                               dtype=self._dtype, \
+                                               non_blocking=True)
+                update_power_dst = update_power.to(state_dev, \
+                                                   dtype=self._dtype, \
+                                                   non_blocking=True)
```

Comment:
D2H copy with a different dtype is always pageable even with `non_blocking=True`. I filed a bug to PyTorch https://github.com/pytorch/pytorch/issues/79933. A temporary workaround is
```python
update_mem_dst = update_mem.to(dtype=self._dtype).to(state_dev, non_blocking=True)
update_power_dst = update_power.to(dtype=self._dtype).to(state_dev, non_blocking=True)
```